OpenTrons/opentrons-api
# feat(app, components): add deck configuration section to device details

# Overview
Add add fixture modal for Desktop and setup instructions modal for Desktop app.
I added the modals for Desktop to the modals that I added for RAUT-663 because of the future modal unification.
I will update the ODD part in following PRs

For the external link href, it hasn't been fixed. (According to Ed)

design
https://www.figma.com/file/nJBVk8a26wdls3ZnkgRmns/OCT-Release%3A-Desktop-%5BDeck-configuration-%26-Drop-tips%5D?node-id=98%3A12072&mode=dev

close RAUT-683

AddDeckConfigurationModal.stories.tsx is broken right now because of QueryClientProvider.
This may be solved by a future PR or removed when ODD part for deck configuration is done.



# Test Plan
1. Turn on deck configuration in FF
2. Go to a Flex's Device details page 
3. Click instructions link button -> the modal is open (the document link isn't update)
4. Click slot add button -> add fixture modal is open and can see an option/options
> A1/B1/C1/D1 the only option is trash bin,
A3/B3/C3 can be trash bin or staging area, and  
D3 can be trash bin, staging area, or waste chute



# Changelog
- update modal components that I added for RAUT-663 to add Desktop versions to them and add tests
- add test for DeviceDetailsDeckConfiguration
- update Stories since the modals' props are updated
- replace QRcode images since they used larger margins and new ones use the min margin  


# Review requests



# Risk assessment
low
